app/forms/search_keywords_form.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SearchKeywordsForm < ApplicationForm
+  attr_reader :current_user, :search_params, :pagination_params
+
+  # rubocop:disable Lint/MissingSuper
+  def initialize(current_user, search_params, pagination_params)
+    @current_user = current_user
+    @search_params = search_params
+    @pagination_params = pagination_params
+  end
+  # rubocop:enable Lint/MissingSuper
+
+  def search_keywords
+    return empty_pagination unless search_params[:adwords_url_contains]
+
+    keywords_has_ads_top_urls_contains_word
+  end
+
+  private
+
+  def keywords_has_ads_top_urls_contains_word
+    pagination, paginated_keywords = pagy(keywords_query.call, pagination_params)
+    word_params = search_params[:adwords_url_contains]
+    filtered_paginated_keywords = paginated_keywords.map { |item| filter_unmatched_adword_urls(item, word_params) }
```

Comment:
I would recommend moving this into a presenter object:
- the presenter can have the `word_params` in its instance variable
- the presenter can expose a new method `matching_adword_urls` that will handle a logic similar to `filter_unmatched_adword_urls` without mutating the `keyword.ads_top_urls`. In general, we should not edit the value or a model field, especially if the resulting value should have a different name (e.g. `matching_adword_urls`).

As a quick draft (non-tested), it would look like this:

```
class KeywordPresenter
  # All delegations needed here...
  delegate :name, ..., to: :keyword

  def initialize(keyword)
    @keyword = keyword
  end

  def matching_adword_urls(word_param)
    keyword.ads_top_urls.select { |item| item.downcase.include?(word_params.downcase) }
  end

  private

  attr_reader :keyword
end
```


> Note that using a presenter implies not using a form :harold: If we need this logic in both the `index` (query object and presenter) and a form (like `edit`), we would need to move that method into a decorator object.